docs/database-structure.md

```diff
@@ -0,0 +1,336 @@
+
+Columns:
+- status
+Possible values:
+  - `online`
+  - `offline`
+  - `draft`
+
+### object_type
+
+Columns:
+- type
+Possible values:
+  - `onboarding`
+  - `campus`
+  - `exercise`
+  - `quest`
+  - `signup`
+  - `exam`
+  - `raid`
+  - `project`
+  - `piscine`
+
+### progress
+
+It is a register of the steps in the progression of students doing the exercises and projects.
+
+Columns:
+- id
+- createdAt
+- updateAt
+- userId
+- groupId
+- eventId
+- version
+- grade
+- isDone
+- path
+- campus
+- objectId
+
+### record
+
+Keeps a register for the bans. It can also keep the record of other situations, this is done by setting the banEndAt to a date already passed and the reason for the record.
+
+Columns:
+- id
+- userId
+- authorId
+- message
+- banEndAt
+- createdAt
+
+### registration
+
+Registrations to events (For example, an exam).
+
+Columns:
+- id
+- createdAt, When the registration was created.
+- startAt, When users can start registering to an event.
+- endAt, When the registration ends.
+- eventStartAt, When the event the registration refers to starts.
+- objectId, The object the registration refers to.
+- parentId, The parent object of the object the registration refers to.
+- attrs
+- path
+- campus
+
+### registration_user
+
+The user registered to events. A user can register to many events and and event can have many users.
+
+Columns:
+- id
+- registrationId
+- userId
+- createdAt
+
+### result
+
+The result for each exercise and project made in the platform.
+
+Columns:
+- id
+- createdAt
+- updatedAt
+- grade
+- progressId
+- attrs
+- type, For the possible types see the table [result_type](#result_type)
+- userId
+- groupId
+- objectId
+- path
+- version, The version of the code the result refers to.
+- eventId
+- isLast, Used to represent transitory states. A progress is only considered finished when `isLast` is `true`.
+- campus
+
+### result_type
+
+Columns:
+- type
+  - `tester`
+  - `user_audit`
+  - `admit_audit`
+  - `admin_selection`
+  - `status`
+
+### role
+
+Columns:
+- id
+- slug
+- name
+- description
+- createdAt
+- updatedAt
+
+### token
+
+Columns:
+- id
+- status
+- createdAt
+- updatedAt
+
+### token_status
+
+Columns:
+- status
+  - `active`
+  - `expired`
+
+### transaction
+
+It's a register of the rewards given to students.
+
+Columns:
+- id
+- type, See [transaction_type](#transaction_type) for the possible types.
+- amount:
+  - type = xp; The amount of Xp that an object rewards.
+  - type = up || down; a percentage of the change (10%);
+- userId
+- attrs
+- createdAt
+- path
+- objectId
+- eventId
+- campus
+
+### transaction_type
+
+Columns:
+- type
+  - `xp`: transaction giving xp
+  - `up`: transaction correspondent to reviewing someone
+  - `down`: transaction correspondent to reviewing you
+
+### user
+
+Columns:
+- id
+- githubId: (Now gitea).
+- githubLogin: (Now gitea).
+- discordId
+- discordLogin
+- profile
+- attrs: Extra information about the users (email, address, etc).
```

Comment:
I would like to define some key attributes (like email and name) and move them to the user directly

Reply:
email for sure - name is related to the name filled in gitea isn't it? because for firstNamee, lastName, these are attributes defined in the forms and it could be written differently by school... to be discussed, I don't know how to prevent it